src/hooks/locale/locale.js

```diff
@@ -0,0 +1,27 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { getLocale } from './selectors'
+
+const fallback = 'en'
+
+const query = {
+    userSettings: {
+        resource: 'userSettings',
+    },
+}
+
+const useLocale = () => {
+    const { loading, error, data } = useDataQuery(query)
+
+    if (loading) {
+        return ''
+    }
+
+    // Fall back to 'en' in case of errors
+    if (error) {
+        return fallback
+    }
+
+    return getLocale(data.userSettings) || fallback
```

Comment:
This ties in with the comment I made about `src/components/App/App.js`:
- Not sure if falling back to `en` on an error is such a great idea. Perhaps it does make sense...
- Maybe instead of returning a string, this hook could return an object similar to `useDataQuery` with `loading` `error` and `locale` (string).

Reply:
Yeah in that case I could just use the data query hook directly. The reason I'm doing this is because we're going to have to render something if there's no locale. English seemed like a reasonable fallback to me.

Reply:
By the way, I should note that this locale is only used to localize the cron translation. And that defaults to english. So what I'm doing here just makes the existing default explicit.

Reply:
Aha, OK makes sense